# Analyzing food and drug interactions through PubMed abstracts

In [2]:
import pubmed.downloader as pb

The _pubmed.downloader_ module implements a _PubMedQuery_ class which searches the PubMed database for a search term and returns at most _max_\__results_ abstracts. 

In [3]:
search_term = 'ACE inhibitor'
max_results = 3
query = pb.PubMedQuery(search_term, max_results)

After instantiating a _PubMedQuery_ instance, _PubMedQuery.id_\__getter_ should be called to read the ids of the articles of the search results.

In [4]:
ids = query.id_getter()
print ids

28573254,28571891,28567349


In [5]:
query.abstract_getter(ids)

{u'28567349': 'The presence of proline at the C-terminal tripeptide of ACE inhibitor can competitively inhibit the ACE activity. The effects of other amino acids are less studied leading to difficulties in predicting potent peptide sequences. The broad specificity of the enzyme may be due to the dual active sites observed on the somatic ACE. The inhibitors may not necessarily competitively inhibit the enzyme which explains why some reported inhibitors do not have the common ACE inhibitor characteristics. Finally, the in vivo assay has to be carried out before the peptides as the antihypertensive agents can be claimed. The peptides must be absorbed into circulation without being degraded, which will affect their bioavailability and potency. Thus, peptides with strong in vitro IC50 values do not necessarily have the same effect in vivo and vice versa.',
 u'28571891': 'Literature search was performed within the PubMed, ScienceDirect.com and Google Scholar.',
 u'28573254': 'The commerciall

Currenty, the abstract getter implementation is a little bit buggy... the abstract above should be one abstract with three fields such as background, methods and results. 

This is not an issue with most of the articles as the abstract is saved within one field..but there a few tricky ones such as ID 28573254... 

However, if we care only about building vocabulary from the text and not if the abstracts are correctly paired with the IDs, then this is fine.

If we went through the IDs one-by-one, the ids could get paired correctly with the abstracts...but then it would take very long.

## Downloading abstracts to local jsons

The _download_\_all_\__abstracts_ method saves _max_\__results_ abstracts to a json file as below.

In [ ]:
# reset the counter before calling the download_all_abstracts method as we have already called it once above
pb.PubMedQuery.COUNT = 0
max_results = 500
pb.download_all_abstracts(search_term, max_results)

Saving to pbabstract1.json
500/50406 downloaded
Saving to pbabstract2.json
1000/50406 downloaded
Saving to pbabstract3.json
1500/50406 downloaded
Saving to pbabstract4.json
2000/50406 downloaded
Saving to pbabstract5.json
2500/50406 downloaded
Saving to pbabstract6.json
3000/50406 downloaded
Saving to pbabstract7.json
3500/50406 downloaded
Saving to pbabstract8.json
4000/50406 downloaded
Saving to pbabstract9.json
4500/50406 downloaded
Saving to pbabstract10.json
5000/50406 downloaded
Saving to pbabstract11.json
5500/50406 downloaded
Saving to pbabstract12.json
6000/50406 downloaded
Saving to pbabstract13.json
6500/50406 downloaded
Saving to pbabstract14.json
7000/50406 downloaded
Saving to pbabstract15.json
7500/50406 downloaded
Saving to pbabstract16.json
8000/50406 downloaded
Saving to pbabstract17.json
8500/50406 downloaded
Saving to pbabstract18.json
9000/50406 downloaded
Saving to pbabstract19.json
9500/50406 downloaded
Saving to pbabstract20.json
10000/50406 downloaded
Saving to